testing/e2e/app/routes/ComboBox/spec.ts

```diff
@@ -0,0 +1,175 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('ComboBox', () => {
+  test('should support keyboard navigation when virtualization is enabled', async ({
+    page,
+  }) => {
+    await page.goto('/ComboBox?virtualization=true');
+
+    await page.keyboard.press('Tab');
+    const comboBoxInput = page.locator('#test-component').locator('input');
+    await expect(comboBoxInput).toHaveAttribute(
+      'aria-controls',
+      'test-component-list',
+    );
+
+    const items = page.getByRole('option');
+
+    //focus first
+    await page.keyboard.press('ArrowDown');
+    await expect(comboBoxInput).toHaveAttribute(
+      'aria-activedescendant',
+      'test-component-option-Item-0',
+    );
+    await expect(items.first()).toHaveAttribute('data-iui-focused', 'true');
+
+    //stay on first
+    await page.keyboard.press('ArrowUp');
+    await expect(comboBoxInput).toHaveAttribute(
+      'aria-activedescendant',
+      'test-component-option-Item-0',
+    );
+    await expect(items.first()).toHaveAttribute('data-iui-focused', 'true');
+
+    //focus second
+    await page.keyboard.press('ArrowDown');
+    await expect(comboBoxInput).toHaveAttribute(
+      'aria-activedescendant',
+      'test-component-option-Item-1',
+    );
+    await expect(items.first()).not.toHaveAttribute('data-iui-focused', 'true');
+    await expect(items.nth(1)).toHaveAttribute('data-iui-focused', 'true');
+
+    //focus last
+    for (let i = 0; i <= 6; ++i) {
+      await page.keyboard.press('ArrowDown');
+    }
+    await expect(comboBoxInput).toHaveAttribute(
+      'aria-activedescendant',
+      'test-component-option-Item-11',
+    );
+    await expect(items.last()).toHaveAttribute('data-iui-focused', 'true');
+
+    //stay on last
+    await page.keyboard.press('ArrowDown');
+    await expect(comboBoxInput).toHaveAttribute(
+      'aria-activedescendant',
+      'test-component-option-Item-11',
+    );
+    await expect(items.last()).toHaveAttribute('data-iui-focused', 'true');
+
+    //select last
+    await page.keyboard.press('Enter');
```

Comment:
Lines checking for the [aria-controls attribute](https://github.com/iTwin/iTwinUI/blob/bf2b3e3b1a47d50210f5d6b3b24944af67b3d98c/testing/e2e/app/routes/ComboBox/spec.ts#L65) were meant to check if the menu was open or not, but that relies on the `aria-controls` value being correctly setup. Added additional check for the presence of the `ComboBoxMenu` by looking for elements with the `listbox` role in this [commit](https://github.com/iTwin/iTwinUI/pull/2061/commits/bf2b3e3b1a47d50210f5d6b3b24944af67b3d98c).

Reply:
Replace `toHaveCount` with `toBeVisible` in this [commit](https://github.com/iTwin/iTwinUI/pull/2061/commits/02f43e165bf52fae15dcde9036c8077b103e1ada). `toBeTruthy`/`toBeFalsy` doesn't seem to work when evaluating locator, and I think that is because the locator is still defined, even if the element that it is pointing to is no longer being rendered.